tardis/tests/tests_slow/report.py

```diff
@@ -0,0 +1,251 @@
+==========
+1. "Writing Plugins" ( https://pytest.org/latest/writing_plugins.html )
+2. "Hookspec Source" ( https://pytest.org/latest/_modules/_pytest/hookspec.html )
+3. "pytest-html" ( https://www.github.com/davehunt/pytest-html )
+"""
+
+from datetime import datetime
+import tempfile
+import time
+
+from pytest_html.plugin import HTMLReport
+from astropy.tests.helper import remote_data
+import tardis
+
+# For specifying error while exception handling
+from socket import gaierror
+
+try:
+    import dokuwiki
+except ImportError:
+    dokuwiki_available = False
+else:
+    dokuwiki_available = True
+
+
+@remote_data
+class DokuReport(HTMLReport):
+
+    def __init__(self, dokuwiki_details):
+        """
+        Initialization of a DokuReport object and registration as a plugin
+        occurs in `pytest_configure`, where a dict containing url, username and
+        password of dokuwiki is passed through `dokuwiki_details`.
+        """
+        logfile = tempfile.TemporaryFile()
+        super(DokuReport, self).__init__(logfile)
+        self.logfile = logfile
+
+        # This list stores tuples of name, result, duration of each test in
+        # dokuwiki syntax strings.
+        self.test_logs = []
+        # This list stores strings containing failure log (if any)
+        # If not, empty strings are appended.
+        self.error_logs = []
+        # This list contains dokuwiki formatted strings of links to comparison
+        # plot images (if any). If not, empty strings are appended.
+        self.plot_images = []
+        # Length of all three lists are same.
+
+        # `generated` stores the date and time of generation of report.
+        # It is set in `pytest_sessionfinish` using datetime.now()
+        self.generated = None
+
+        # File path of image of comparison plots for a single test.
+        self.image_path = None
+
+        # Make a dokuwiki connection if dokuwiki import was successful.
+        if dokuwiki_available:
+            try:
+                self.doku_conn = dokuwiki.DokuWiki(
+                    url=dokuwiki_details["url"],
+                    user=dokuwiki_details["username"],
+                    password=dokuwiki_details["password"])
+            except gaierror, dokuwiki.DokuWikiError:
+                print "Dokuwiki connection not established, report upload failed!"
+                self.doku_conn = None
+
+    def add_plot(self, path):
+        """
+        This method is used by any single test method to add the file path of
+        image of corresponding comparison plot.
+        """
+        self.image_path = path
+
+    def pytest_sessionstart(self, session):
+        """
+        This hook function is called by pytest after all the methods and
+        collected and before the test execution has been started. Here, the
+        starting time is being recorded.
+        """
+        super(DokuReport, self).pytest_sessionstart(session)
+
+    def pytest_runtest_logreport(self, report):
```

Comment:
These are pytest hooks, this class is registered as a plugin so pytest automatically calls these hook functions. I'll update this PR by their removal if removing them also works (it should, according to inheritance).